x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts

```diff
@@ -4,15 +4,36 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { AlertData, ImmutableArray } from '../../../../../common/types';
+// import { AlertData, ImmutableArray } from '../../../../../common/types';
+import qs from 'querystring';
 import { AppAction } from '../action';
 import { MiddlewareFactory } from '../../types';
 
 export const alertMiddlewareFactory: MiddlewareFactory = coreStart => {
+  interface Params {
+    paging_properties: object[];
+  }
+
+  const params: Params = {
+    paging_properties: [],
+  };
+
+  const qp = qs.parse(window.location.search.slice(1));
+  if ('page_size' in qp && typeof qp.page_size === 'string') {
+    const pageSize: number = parseInt(qp.page_size, 10);
+    params.paging_properties.push({ page_size: pageSize });
+  }
+  if ('page_index' in qp && typeof qp.page_index === 'string') {
+    const pageIndex: number = parseInt(qp.page_index, 10);
```

Comment:
if you want to be super cool, you could check if `pageIndex` is `NaN` before using it:

`isNaN(pageIndex)`. `NaN` values will be returned from `parseInt` in the case of a string that can't be parsed.

Reply:
This should be resolved through the schema now. Removed the parseInt calls.